From 98f3041e2442e2c8ddf18e07ce9c24367312379f Mon Sep 17 00:00:00 2001
From: Markus Niebel <Markus.Niebel@tq-group.com>
Date: Wed, 30 Jan 2019 16:26:11 +0100
Subject: [PATCH] arm64: dt: fsl-imx8mq-tqma8mq: change memory node

use smallest variant planned for TQMa8MQ modules
while at it, adopt linux,cma node to this setting

Signed-off-by: Markus Niebel <Markus.Niebel@tq-group.com>
---
 .../arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi | 12 ++++++++++--
 1 file changed, 10 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
index af9654eaa5a3..3d25c3c802b9 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mq-tqma8mq.dtsi
@@ -34,6 +34,12 @@
 		regulator-always-on;
 	};
 
+	memory@40000000 {
+		device_type = "memory";
+		/*  our minimum RAM config will be 512 MiB */
+		reg = <0x00000000 0x40000000 0 0x20000000>;
+	};
+
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
@@ -43,8 +49,10 @@
 		linux,cma {
 			compatible = "shared-dma-pool";
 			reusable;
-			size = <0 0x10000000>;
-			alloc-ranges = <0 0x40000000 0 0x40000000>;
+			/* 256 + 128 MiB = 384 */
+			size = <0 0x18000000>;
+			/*  512 - 64 MiB, our minimum RAM config will be 512 MiB */
+			alloc-ranges = <0 0x40000000 0 0x1C000000>;
 			linux,cma-default;
 		};
 	};
